(fix) booking algorithm

teetangh opened the pull request:
- calendar slot timings for 4 events

Divyam-Kumar-Pandey committed:
Enhance consultant dashboard and API functionality

- Add thin scrollbar styling to globals.css
- Comment out verbose console logs in onboarding API route for cleaner output
- Simplify RequestsPage component by removing unnecessary parameter handling
- Update RequestSlotAllocationTab to make onUpdate prop optional and improve slot allocation logic
- Modify Calendar component to support half-hour intervals and enhance time slot handling
- Introduce utility functions for converting UTC to local time and rounding timestamps
- Update appointment creation logic to round appointment times to the nearest 30 minutes

teetangh commented on app/dashboard/consultant/[consultantId]/(features)/requests/page.tsx:
why are we not requiring params here ?

teetangh commented on app/dashboard/consultant/[consultantId]/(features)/requests/page.tsx:
why are we not requiring params here ?

teetangh commented on app/dashboard/consultant/[consultantId]/(features)/requests/page.tsx:
```diff
@@ -1,19 +1,8 @@
-}>) {
-  const resolvedParams = use(params);
-  const consultantId = resolvedParams.consultantId;
+export default function RequestsPage() {
```
why are we not requiring params here ?

Divyam-Kumar-Pandey replied:
I changed this while I was fixing the re-rendering issue.
This part was unnecessary, and I thought this might be causing some issue, so I removed it.

Divyam-Kumar-Pandey replied:
But I don't think this is causing some issue, so we can revert this if you want.

Divyam-Kumar-Pandey committed:
Monthly view in Calender, showing available slots and selection

Divyam-Kumar-Pandey committed:
fix: re-rendering issue on consultee dashboard

Divyam-Kumar-Pandey committed:
fix: update Calendar component to handle weekly and monthly schedule selection

- Adjust logic for determining existing and selected slots based on schedule type
- Modify onClick handlers to use UTC string for slot selection

Divyam-Kumar-Pandey committed:
refactor: remove past slot validation from allocation routes

Divyam-Kumar-Pandey committed:
fix: update AnnouncementBar visibility logic

Divyam-Kumar-Pandey committed:
fix: update appointment handling in API and enhance event display

- Modify PATCH route to use eventId instead of appointmentId for appointment updates
- Add console logs for debugging appointment retrieval and update processes
- Enhance EventCarousel component to display event start and end times
- Improve TimingsCalendar component with better overflow handling and scrollbar styling
- Add console logs for existing response data in TimingsCalendar

teetangh committed:
fix: update appointment API to use appointmentId and enhance error handling

teetangh commented:
/gemini review
